tests/scripts/pulp_python/test_repository.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+# shellcheck source=tests/scripts/config.source
+. "$(dirname "$(dirname "$(realpath "$0")")")"/config.source
+
+pulp debug has-plugin --name "pulp_python" --min-version "3.1.0.dev" || exit 3
+
+cleanup() {
+  pulp python repository destroy --name "cli_test_python_repo" || true
+  pulp python remote destroy --name "cli_test_python_remote1" || true
+  pulp python remote destroy --name "cli_test_python_remote2" || true
+}
+trap cleanup EXIT
+
+REMOTE1_HREF="$(pulp python remote create --name "cli_test_python_remote1" --url "http://a/" | jq -r '.pulp_href')"
+REMOTE2_HREF="$(pulp python remote create --name "cli_test_python_remote2" --url "http://b/" | jq -r '.pulp_href')"
+
+expect_succ pulp python repository list
+
+expect_succ pulp python repository create --name "cli_test_python_repo" --description "Test repository for CLI tests"
+expect_succ pulp python repository update --name "cli_test_python_repo" --description "" --remote "cli_test_python_remote1"
+expect_succ pulp python repository show --name "cli_test_python_repo"
+expect_succ test "$(echo "$OUTPUT" | jq -r '.remote')" = "$REMOTE1_HREF"
+expect_succ pulp python repository update --name "cli_test_python_repo" --remote "cli_test_python_remote2"
+expect_succ pulp python repository show --name "cli_test_python_repo"
+expect_succ test "$(echo "$OUTPUT" | jq -r '.remote')" = "$REMOTE2_HREF"
+expect_succ pulp python repository update --name "cli_test_python_repo" --remote ""
+expect_succ pulp python repository show --name "cli_test_python_repo"
+expect_succ test "$(echo "$OUTPUT" | jq -r '.description')" = "null"
+expect_succ test "$(echo "$OUTPUT" | jq -r '.remote')" = ""
+expect_succ pulp python repository list
+test "$(echo "$OUTPUT" | jq -r '.|length')" != "0"
+if pulp debug has-plugin --name "pulpcore" --min-version "3.10.dev"
+then
+  expect_succ pulp repository list
+  test "$(echo "$OUTPUT" | jq -r '.|length')" != "0"
+  expect_succ pulp repository list --name "cli_test_python_repo"
+  test "$(echo "$OUTPUT" | jq -r '.|length')" = "1"
+  expect_succ pulp repository list --name-contains "cli_test_python"
+  test "$(echo "$OUTPUT" | jq -r '.|length')" = "1"
+  expect_succ pulp repository list --name-icontains "CLI_test_python"
+  test "$(echo "$OUTPUT" | jq -r '.|length')" = "1"
+  expect_succ pulp repository list --name-in "cli_test_python_repo"
+  test "$(echo "$OUTPUT" | jq -r '.|length')" = "1"
+fi
```

Comment:
This looks like test for core, i think we should remove it here.

If you took this from the corresponding file test, i admit bad design on my behalf.